pkg/openstack/test.go

```diff
@@ -0,0 +1,77 @@
+package openstack
+
+import (
+	"context"
+	"time"
+
+	"github.com/openstack-k8s-operators/lib-common/modules/common/condition"
+	"github.com/openstack-k8s-operators/lib-common/modules/common/configmap"
+	"github.com/openstack-k8s-operators/lib-common/modules/common/helper"
+	"github.com/openstack-k8s-operators/lib-common/modules/common/util"
+
+	corev1beta1 "github.com/openstack-k8s-operators/openstack-operator/apis/core/v1beta1"
+	k8s_errors "k8s.io/apimachinery/pkg/api/errors"
+	ctrl "sigs.k8s.io/controller-runtime"
+)
+
+const (
+	// config map generated by Keyston operator containing clouds.yaml
+	CloudsConfigMap       = "openstack-config"
+	TestOperatorConfigMap = "test-operator-config"
+
+	TempestImageKey = "tempest-image"
+	TobikoImageKey  = "tobiko-image"
+	CloudsKey       = "clouds-secret-name"
+)
+
+func ReconcileTest(ctx context.Context, instance *corev1beta1.OpenStackControlPlane, version *corev1beta1.OpenStackVersion, helper *helper.Helper) (ctrl.Result, error) {
+	customData := map[string]string{
+		TempestImageKey: *version.Status.ContainerImages.TestTempestImage,
+		TobikoImageKey:  *version.Status.ContainerImages.TestTobikoImage,
+		CloudsKey:       CloudsConfigMap,
+	}
+
+	_, _, err := configmap.GetConfigMapAndHashWithName(ctx, helper, CloudsConfigMap, instance.Namespace)
```

Comment:
Why do we wait on the Clouds config map here? I guess it is because test-operator needs it to functionally work. In that case should the wait occur in that operator instead?

Reply:
Thank you @dprince for the review!

As I'm reading this now it would probably make a more sense to move this waiting into the test-operator. I'm going to update the PR.

Reply:
@dprince the wait is removed.